app/poi_api/poi_suggestion.py

```diff
@@ -1,28 +1,62 @@
 from datetime import datetime
+from app.poi_api.poi_errors import POINotFoundError, InvalidPOISuggestionError
+from app import common
 
 
-class POI_suggestions:
+class POI_suggestions(common.FirebaseDataEntity):
     def __init__(
-        self, pid, suggestion_name, notes, submission_time, submitted_by
+        self,
+        db_ref,
+        pid,
+        suggestion_name,
+        notes,
+        submitted_by,
+        submission_time=datetime.now(),
     ) -> None:
-        self.pid = pid
+        super().__init__(db_ref)
+        self._pid = pid
         self.suggestion_name = suggestion_name
         self.notes = notes
-        self.submission_time = submission_time
         self.submitted_by = submitted_by
+        self.submission_time = submission_time
 
     def get_pid(self):
-        return self.pid
+        return self._pid
+
+    def get(db_ref, id):
+        target_data = db_ref.document(id).get()
+        if not target_data.exists:
+            raise POINotFoundError(id)
+        return POI_suggestions.from_dict(db_ref, target_data.to_dict())
 
     def to_dict(self):
         poi_suggestion_dict = {
-            "_pid": self.pid,
+            "_pid": self._pid,
             "suggestion_name": self.suggestion_name,
             "notes": self.notes,
             "submission_time": self.submission_time,
             "submitted_by": self.submitted_by,
         }
         return poi_suggestion_dict
 
-    def from_dict(self, poi_suggestion):
-        pass
+    def from_dict(db_ref, dict):

```

Comment:
I had the same issue before I fixed it in my last PR, but I think we should be using dict.get() with a default value for all parameters that are not required. This way, even if there is a parameter that is missing (for example, we add something later but there is still old data without the new parameter), it can still be parsed